src/pages/editProfile/index.js

```diff
@@ -0,0 +1,229 @@
+import { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+import TextInput from "../../components/form/textInput";
+import Button from "../../components/button";
+import ProfileCircle from "../../components/profileCircle";
+import Card from "../../components/card";
+import { useNavigate } from "react-router-dom";
+import useModal from "../../hooks/useModal";
+import ErrorMessage from "../../components/errorMessage";
+import "./style.css";
+import SaveChangesModal from "../../components/saveChangesModal"
+import { get } from "../../service/apiClient";
+import { emptyProfile } from "../../service/mockData";
+import useAuth from "../../hooks/useAuth";
+import ProfileImg from "../../components/profileImage/ProfileImage";
+
+
+const EditProfile = () => {
+  const [profile, setProfile] = useState(emptyProfile);
+  const [formState, setFormState] = useState(profile);
+  const [readOnly, setReadOnly] = useState(false)
+  const [passwordPermission, setPasswordPermission] = useState(true)
+  const [isError, setIsError] = useState(false);
+  const navigate = useNavigate()
+  const { id } = useParams()
+  const { openModal, setModal } = useModal();
+  const { loggedInUserInfo } = useAuth();
+
+ 
+
+  const showModal = (event) => {
+    event.preventDefault()
+    setModal(<SaveChangesModal loggedInUserInfo={loggedInUserInfo} id={id} formState={formState} />);
+    openModal();
+  };
+  const cancelChanges = () => {
+    navigate(-1)
+  }
+
+  const handleChange = (event) => {
+    const value = event.target.value
+    const name = event.target.name
+    const newFormState = { ...formState }
+    newFormState[name] = value
+    setFormState(newFormState)
+  }
+  const profileData = async () => {
+    const res = await get(`users/${id}`);
+    res.data.user ? setFormState(res.data.user) : setIsError(true);
+    res.data.user ? setProfile(res.data.user) : setIsError(true)
+  };
+
+  const controlPagePermission = () => {
+    if (loggedInUserInfo.role === "STUDENT" && loggedInUserInfo.id !== Number(id)) {
+      navigate(-1)
+    }
+    if (loggedInUserInfo.role === "STUDENT") {
+      setReadOnly(true)
+    }
+    if (loggedInUserInfo.id === Number(id)) {
+      setPasswordPermission(false)
```

Comment:
I would expect if I am the user to be able to change my password to set passwword permissions to true